### ISSUE TITLE
WebView crashes on second text selection [Android]

### ISSUE THREAD

**Issue by amirh** [flutter/flutter#21239]
It looks like the selection handles are not removed from the window, and are being re-added on a second text selection which crashes the app with:

```
W/System.err(30171): java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.                                                            
W/System.err(30171):    at android.view.ViewGroup.addViewInner(ViewGroup.java:4309)                                                                                                                                
W/System.err(30171):    at android.view.ViewGroup.addView(ViewGroup.java:4145)                                                                                                                                     
W/System.err(30171):    at android.view.ViewGroup.addView(ViewGroup.java:4103)                                                                                                                                     
W/System.err(30171):    at android.widget.PopupWindow.createDecorView(PopupWindow.java:1233)                                                                                                                       
W/System.err(30171):    at android.widget.PopupWindow.preparePopup(PopupWindow.java:1177)                                                                                                                          
W/System.err(30171):    at android.widget.PopupWindow.showAtLocation(PopupWindow.java:1022)                                                                                                                        
W/System.err(30171):    at android.widget.PopupWindow.showAtLocation(PopupWindow.java:995)                                                                                                                         
W/System.err(30171):    at org.chromium.android_webview.PopupTouchHandleDrawable.show(PopupTouchHandleDrawable.java:237)                                                                                           
W/System.err(30171):    at org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)                                                                                                               
W/System.err(30171):    at org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:9)                                                                                                       
W/System.err(30171):    at android.os.Handler.dispatchMessage(Handler.java:102)                                                                                                                                    
W/System.err(30171):    at android.os.Looper.loop(Looper.java:148)                                                                                                                                                 
W/System.err(30171):    at android.app.ActivityThread.main(ActivityThread.java:5417)                                                                                                                               
W/System.err(30171):    at java.lang.reflect.Method.invoke(Native Method)                                                                                                                                          
W/System.err(30171):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)                                                                                                         
W/System.err(30171):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)               
```

Temporarily we prevent text selection in WebView to avoid the crash.

**Comment by defaultbr** [comment]
i've removed the part  `onLongPress: () {},` from webview_flutter.dart and its not crashing.

i can saw that the styled selected text is working but the thumb to drag are not (but the physical space is reserved, since its possible to click where the thumb should be and drag the selection (working).

Its not drawing the thumb because a Activity Context is missing? 

Maybe is possible on delegate the onActionModeStarted/onActionModeFinished to the flutter part?

My intetion is to create a custom rich text where you can select the text (expand or reduce the selection) and select bold/italic etc.. in the floating menu above selection, than it passes the javascript command to wrap the selected into a span with custom class

if i get some news/progress i will post here

**Comment by amirh** [comment]
The crash was fixed by #6266, so I'm closing this issue.

Filed #24584 for the selection handles.
